Dockerfile

```diff
@@ -1,6 +1,14 @@
 FROM registry.svc.ci.openshift.org/openshift/release:golang-1.12 AS builder
 WORKDIR /go/src/github.com/openshift/cluster-logging-operator
-COPY . .
+
+COPY cmd ./cmd
+COPY manifests ./manifests
+COPY pkg ./pkg
+COPY vendor ./vendor
+COPY version ./version
+COPY Dockerfile ./Dockerfile
+COPY Makefile ./Makefile
+
```

Comment:
Ordering and grouping was also my proposal back then when `Dockerfile.fast` was introduced:
```
#
# Order of COPY statements based on frequency of change.
# This enables better caching of each layer.
#

# Configuration
COPY config config
COPY files files

# manifests
COPY manifests manifests
COPY olm_deploy olm_deploy

# vendored dependencies
COPY Gopkg.lock Gopkg.lock
COPY Gopkg.toml Gopkg.toml
COPY vendor vendor

# source code
COPY cmd cmd
COPY pkg pkg
COPY scripts scripts
COPY version version

# make targets
COPY Makefile Makefile
RUN make build
```

Reply:
Looks like @periklis has identified additional directories to copy

Reply:
You can take my list if you like. I am working with this based on git stash for weeks now. The order of the groups rely on the frequency of change and dependency order (e.g. a manifests change means rebuild most of the code because CRD go and yaml files changed)

Reply:
we dont need all the directories to be copied. The list i have is sufficient for building the executable at the `RUN make build` step 
other directories are getting copied later in the Dockerfile.

Reply:
should the order matter? as of now i see that it maters. As soon as there is a step which has a change, buildah stops using cache for steps which appear later, even though it is unrelated.

Reply:
The list is missing the `scripts`. Because `make build` works it doesn't tell that the operator works. We need the `scripts` dir for the `cert_generation.sh`.

Reply:
In addition the order provide in the PR is akward. If `pkg` changes the big `vendor` will be rebuild again.

Reply:
Furthermore how can your Dockerfile work without the dirs `files` and `manifests`. They are used in the next build stage.

Reply:
`scripts` are being copied later than the `make build` step
updated the order
`files` are being copied later than the `make build` step. `manifests` copying is there

Reply:
:hear_no_evil: by mistake, you made the copies directly in the runner build stage. IMHO this is not bad, but also not the best approach. One reason why the second stage was using `--from` in COPY statements is to make the build context only attached to the first stage. So that subsequent build stages only relate to artifacts on the cached layers of the previous ones.

Reply:
for `manifests` and outputs of `make build`, it makes sense to use `--from` COPY statements, as they are produced/changed in first stage.
for copying from filesystem, is it not ok to use plain COPY statements?